server/konnectors/currencies.coffee

```diff
@@ -0,0 +1,152 @@
+
+# Models
+
+#Currency = americano.getModel 'Currency',
+#    code: String # ISO 4217 currency code
+#    title: String
+#    symbol: String
+
+CurrencyRate = americano.getModel 'CurrencyRate'
+    date: Date
+    rate: Number
+    currency: String
+    base: String
+
+CurrencyRate.all = (callback) ->
+    CurrencyRate.request 'byDate', callback
+
+# Konnector
+
+module.exports =
+
+    name: "Currencies"
+    slug: "currencies"
+    description: 'konnector description currencies'
+    vendorLink: "https://www.ecb.europa.eu/"
+
+    fields:
+        EUR: "checkbox"
+        USD: "checkbox"
+        JPY: "checkbox"
+        BGN: "checkbox"
+        CZK: "checkbox"
+        DKK: "checkbox"
+        GBP: "checkbox"
+        HUF: "checkbox"
+        PLN: "checkbox"
+        RON: "checkbox"
+        SEK: "checkbox"
+        CHF: "checkbox"
+        NOK: "checkbox"
+        HRK: "checkbox"
+        RUB: "checkbox"
+        TRY: "checkbox"
+        AUD: "checkbox"
+        BRL: "checkbox"
+        CAD: "checkbox"
+        CNY: "checkbox"
+        HKD: "checkbox"
+        IDR: "checkbox"
+        ILS: "checkbox"
+        INR: "checkbox"
+        KRW: "checkbox"
+        MXN: "checkbox"
+        MYR: "checkbox"
+        NZD: "checkbox"
+        PHP: "checkbox"
+        SGD: "checkbox"
+        THB: "checkbox"
+        ZAR: "checkbox"
+    models:
+        currencyrate: CurrencyRate
+
+    # Define model requests.
+    init: (callback) ->
+        map = (doc) -> emit doc.date, doc
+        CurrencyRate.defineRequest 'byDate', map, (err) ->
+            callback err
+
+    # Fetch rates data from the ECB website and save them as Cozy objects.
+    fetch: (requiredFields, callback) ->
+        log.info 'import started'
+
+        url = dailyUrl
+        today = new Date() # FIXME day-precision, not millisecond-precision.
+
+        # Compute the latest rates we know.
+        # FIXME maybe this doesn't need to go through all the rates?
+        latestRates = {}
+        CurrencyRate.all, (err, rates) ->
+            return callback err if err
+            for rate in rates
+                date = latestRates[rate.currency].date
+                if !date or date < rate.date
+                    latestRates[rate.currency] = rate
+        
+        # Figure out how far back in time we need to fetch.
+        for currency in requiredFields
+            if !requiredFields[currency] #.checked ?
+                continue
+            date = latestRates[currency].date
+            if !date or date #older than 90days
+                # Jackpot! We have to refetch from the beginning.
+                url = histUrl
+                break
+            if date #older than yesterday
+                url = recentUrl
+
+        options =
+            method: 'GET'
+            url: url
+
+        request options, (err, res, body) ->
+            return callback err if err
+
+            result = parseXMLString body # FIXME use npm module xml2js?
```

Comment:
This should be something like `xml2js.parseString(body, function(err, result) { ... }` when `xml2js` is actually imported.